### ISSUE TITLE
Support for handling/ploting train, test and validation errors versus iterations

### ISSUE THREAD

**Issue by aferust** [aigamedev/scikit-neuralnetwork#189]
In nn toolbox of matlab, you can display training record to reveal any risk of overfitting. Is there any way to access MSE values of training, testing, and validation during training epoches for creating training record such as http://www.mathworks.com/cmsimages/70376_wl_neural_main_wl.jpg?

Edit: Oh, I have found that there are callbacks doing this. tbrittoborges has already pointed this out:
https://github.com/aigamedev/scikit-neuralnetwork/issues/174

**Comment by aferust** [comment]
Ok, if GridSearchCV of sklearn is needed, some more tricks are required to obtain training record related with each tuned_parameters of nn. Here is an axample:

"""Those are definitions for callbacks:"""
valid_err_pack, train_err_pack = [], []
valid_err, train_err = [], []
def get_errors(best_valid_error, best_train_error, **_):
    valid_err.append(best_valid_error)
    train_err.append(best_train_error)
    
def pack_errors(**variables):
    global valid_err
    global train_err
    valid_err_pack.append(valid_err)
    train_err_pack.append(train_err)
    train_err = []
    valid_err = []

dictionary defining a grid space:
tuned_parameters={
    'learning_rate': [0.05, 0.01, 0.005, 0.001],
    #'hidden0__units': [4, 8, 12, 16, 20],
    'hidden0__units': [30],
    'hidden0__type': ["Rectifier", "Sigmoid", "Tanh", "Softmax"]}

An example MLP including appropriate callbacks:
nn = Classifier(
    layers=[
        Layer("Rectifier"),
        Layer("Softmax")],
    learning_rate=0.001,
    n_iter=250,
    random_state=1,
    valid_size=0.1,
    callback={'on_epoch_finish': get_errors, 'on_train_finish': pack_errors}
    )

I hope those will be helpful for ones who needs storing training records for each network combination during grid search.

**Comment by alexjc** [comment]
I'll try to make this more clear in the documentation!
